flytekit/bin/entrypoint.py

```diff
@@ -93,23 +98,40 @@ def _dispatch_execute(ctx: FlyteContext, task_def: PythonTask, inputs_path: str,
             output_file_dict = {_constants.FUTURES_FILE_NAME: outputs}
         else:
             _logging.getLogger().error(f"SystemError: received unknown outputs from task {outputs}")
-            # TODO This should probably cause an error file
-            return
-
-        for k, v in output_file_dict.items():
-            _common_utils.write_proto_to_file(v.to_flyte_idl(), _os.path.join(ctx.execution_state.engine_dir, k))
-
-        # Step3a
-        ctx.file_access.upload_directory(ctx.execution_state.engine_dir, output_prefix)
-        _logging.info(f"Outputs written successful the the output prefix {output_prefix}")
+            output_file_dict[_constants.ERROR_FILE_NAME] = _error_models.ErrorDocument(
+                _error_models.ContainerError(
+                    "UNKNOWN_OUTPUT",
+                    f"Type of output received not handled {type(outputs)} outputs: {outputs}",
+                    _error_models.ContainerError.Kind.RECOVERABLE,
+                )
+            )
+    except _scoped_exceptions.FlyteScopedException as e:
+        _logging.error("!! Begin Error Captured by Flyte !!")
+        output_file_dict[_constants.ERROR_FILE_NAME] = _error_models.ErrorDocument(
+            _error_models.ContainerError(e.error_code, e.verbose_message, e.kind)
+        )
+        _logging.error(e.verbose_message)
+        _logging.error("!! End Error Captured by Flyte !!")
     except Exception as e:
         if isinstance(e, IgnoreOutputs):
             # Step 3b
             _logging.warning(f"IgnoreOutputs received! Outputs.pb will not be uploaded. reason {e}")
             return
         # Step 3c
         _logging.error(f"Exception when executing task {task_def.name}, reason {str(e)}")
-        raise e
+        _logging.error("!! Begin Unknown System Error Captured by Flyte !!")
+        exc_str = _traceback.format_exc()
+        output_file_dict[_constants.ERROR_FILE_NAME] = _error_models.ErrorDocument(
+            _error_models.ContainerError("SYSTEM:Unknown", exc_str, _error_models.ContainerError.Kind.RECOVERABLE,)
+        )
+        _logging.error(exc_str)
+        _logging.error("!! End Error Captured by Flyte !!")
+
+    for k, v in output_file_dict.items():
+        _common_utils.write_proto_to_file(v.to_flyte_idl(), _os.path.join(ctx.execution_state.engine_dir, k))
+
+    ctx.file_access.upload_directory(ctx.execution_state.engine_dir, output_prefix)
```

Comment:
So what happens if this fails?

Reply:
This is the issue we talked about earlier, isn't it? anyway we can capture that and write the original error (+ the new) to stdErr?

Reply:
The exception gets caught at a higher layer and you see the error message you don't like https://nuclyde.slack.com/archives/D01FD1XGYRZ/p1612578571113500